### PR TITLE
#235 - Add display_label field to document type, update index and views

### PR DESCRIPTION
**Pull request by blms**
### What this PR does
- Per #235:
  - Adds optional `display_label` field to `DocumentType` model
  - Updates `DocumentType` `__str__` to use `display_label` if available
  - Adds data migration to add the provided labels
  - Updates all instances of "Unknown" to "Unknown type"
  - Updates `Document` solr index to use the new `__str__` method
  - Fixes tests with duplicate DocumentTypes
  - Adds SASS mixin for properly formatted doctype in titles, used in both search and detail view

### Steps to test

1. Visit the document search page
2. Verify that document titles display in the format "**Document type**; shelfmark"
3. Verify that the document types display according to the labels specified in #235 
4. Verify that documents with unknown type read "Unknown type", rather than just "Unknown"
4. Navigate to a document detail page
5. Verify that 2-4 remain true on this page

### Deploy notes 

- Solr reindexing is required on deploy.

### CODE REVIEW

File: geniza/corpus/templates/corpus/document_detail.html
Comment (rlskoeser) [inline geniza/corpus/templates/corpus/document_detail.html:25]: "unknown type" needs to be marked for translation here and in the model code

File: geniza/corpus/templates/corpus/document_detail.html
Comment (rlskoeser) [inline geniza/corpus/templates/corpus/document_detail.html:26]: This can be simpler too, actually, although I'm not sure how you integrate it with the translation (maybe need to get the translation text first).
```suggestion
         <span class="doctype">{{ document.doctype|default:"Unknown type" }}</span>
```

File: geniza/corpus/tests/test_corpus_models.py
Comment (rlskoeser) [inline geniza/corpus/tests/test_corpus_models.py:215]: What are you testing here?

What you need to test is the new case for `str` when `display_label` is set

File: geniza/corpus/views.py
Comment (rlskoeser) [inline geniza/corpus/views.py:140]: Let's specify document or document-detail, since we could have other kinds of detail pages.

And now that I see it again I'm not sure about using `id` for the attribute here, that seems potentially misleading. Could you come up with something a bit more specific to how this is used?

File: sitemedia/scss/main.scss
Comment (rlskoeser) [inline sitemedia/scss/main.scss:17]: here too, I think document would be clearer than detail

**Review by rlskoeser — CHANGES_REQUESTED** [review]
In general this looks good, just a few things that need a bit of revision.

File: geniza/corpus/migrations/0020_populate_documenttype_display_label.py
Comment (rlskoeser) [inline geniza/corpus/migrations/0020_populate_documenttype_display_label.py:16]: one more: there's no need to set the display label when it is the same as the name, so please drop these two

File: geniza/corpus/models.py
Comment (rlskoeser) [inline geniza/corpus/models.py:530]: We _might_ need to differentiate between the actual document type and the display label for the type — we're going to have a facet/filter on document type and I'm not sure which one makes most sense in that context. Seems reasonable to assume for now that they should be consistent, so let's leave this for now.

File: geniza/corpus/models.py
Comment (blms) [reply geniza/corpus/models.py:224]: Thanks! How do we handle translations for help text, by the way?

Comment (rlskoeser) [reply geniza/corpus/models.py]: We need to start tagging them! I was just reading more about django localization because I had a question about how to test it, and I saw that they recommend you explicitly set model verbose names and make them translatable — which we have _not_ done yet. I think hold off on making this one translatable and I'll add a chore to set it up across all the models (verbose names, help text, etc.)

File: geniza/corpus/views.py
Comment (blms) [reply geniza/corpus/views.py]: I ended up going with `page_type` here, let me know if that doesn't make sense!

File: geniza/corpus/migrations/0019_documenttype_display_label.py
Comment (rlskoeser) [inline geniza/corpus/migrations/0019_documenttype_display_label.py:16]: It irks me that django creates migrations for meta changes like help text, and I try to avoid creating unnecessary migrations, especially within a feature branch.

Please integrate this help text change to the initial migration, then delete 0021 and confirm it's no longer needed.

```suggestion
            field=models.CharField(
                blank=True,
                help_text="Optional label for display on the public site",
                max_length=255,
            ),
```

File: geniza/corpus/models.py
Comment (rlskoeser) [inline geniza/corpus/models.py:495]: 🤔 I wonder if we need to translate "??"

**Review by rlskoeser — APPROVED** [review]
Looks great, thanks for making the changes; `page_type` makes sense to me, much more meaningful than `id`!

Requesting one small change with the migrations, then we can merge.

**Comment by blms** [comment]
All set, thanks @rlskoeser! Let me know also if I should be merging things once all have approved; am I right that the QA workflow is merge, move to "dev complete" in ZenHub, and include testing instructions in the PR (or does that go in the issue)?

**Pull request merged**
